### PR TITLE
Remove distutils

### PR DESCRIPTION
**Pull request by ax3l**
Distutils is gone in Python 3.12+.

Refs.:
- https://peps.python.org/pep-0632/#migration-advice
- https://github.com/AMReX-Codes/pyamrex/pull/180

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,189 +30,7 @@ Our workflow is described in [CONTRIBUTING.rst](CONTRIBUTING.rst).
 
 ## Developer Environment
 
-Please prepare you local development environment as follows.
```
Comment (dpgrote) [inline README.md]: This is fine here, but would be cleaner if this change was done in a separate PR since it isn't related to the main topic.

**Comment by dpgrote** [comment]
This looks good and I can approve, though I have one comment posted.

Would there be a way to put all of the "python -m pip install" commands in a script so that they don't need to be written out in each of the 10 files as is now?

File: README.md (hunk 1 above)
Comment (ax3l) [reply README.md]: Agreed, sorry that was a it sloppy and makes this harder to Backport/patch where needed indeed.

**Comment by ax3l** [comment]
Yes, technically this is all in pyproject.yaml, let me see how to install that in a two liner or so.

Pip has to definitely be upgraded first and solely, because some people start with ancient pip installs that would not understand later pip steps we do.

Update: since we do not build with `--no-build-isolation`, technically all we would need is the latest `pip`... Will revisit that part another time, I have not problem updating this with `git grep` where listed. Avoids that one has to have the source cloned at that step.

File: README.md (hunk 1 above)
Comment (ax3l) [reply README.md]: Moved this to #453
